File: v1/src/main/java/com/api/v1/medical_appointments/domain/MedicalAppointmentAuditTrail.java
package com.api.v1.medical_appointments.domain;

import com.api.v1.medical_appointments.MedicalAppointment;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.UUID;

@Document(collection = "MedicalAppointmentAuditTrail")
public record MedicalAppointmentAuditTrail(
        @Id
        String id,
        MedicalAppointment medicalAppointment,
        LocalDateTime createdAt
) {

    public static MedicalAppointmentAuditTrail of(MedicalAppointment medicalAppointment) {
        return new MedicalAppointmentAuditTrail(
                UUID.randomUUID().toString(),
                medicalAppointment,
                LocalDateTime.now()
        );
    }
}

File: v2/src/main/kotlin/com/api/v2/medical_appointment/domain/MedicalAppointmentAuditTrail.kt
package com.api.v2.medical_appointment.domain

import com.api.v2.medical_appointment.MedicalAppointment
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.LocalDateTime
import java.util.UUID

@Document(collection = "MedicalAppointmentAuditTrail")
data class MedicalAppointmentAuditTrail(
    @Id
    val id: String,
    val medicalAppointment: MedicalAppointment,
    val createdAt: LocalDateTime
) {

    companion object {
        fun of(medicalAppointment: MedicalAppointment): MedicalAppointmentAuditTrail {
            return MedicalAppointmentAuditTrail(
                UUID.randomUUID().toString(),
                medicalAppointment,
                LocalDateTime.now()
            )
        }
    }
}
